### PR TITLE
Add "pool" option and update gitignore

### PR DESCRIPTION
**Pull request by andrew103**
Added the rhsm "pool" option where necessary and updated the .gitignore file to exclude the symlink

### CODE REVIEW

**Review by greg-hellings — APPROVED** [review]

File: .gitignore [hunk 1]
```diff
@@ -3,3 +3,4 @@
 *.pyc
 *.pyo
 __pycache__/*
+basic_subscription
```
Comment (tehsmyers) [inline .gitignore]: This is a downstream component; upstream should have no knowledge of downstream. Please remove it.

File: README.md [hunk 2]
```diff
@@ -38,6 +38,7 @@ Vars in this section directly correspond to the args available to the
 * `rhsm_server_proxy_password` - HTTP proxy password
 * `rhsm_auto_attach` - automatically consume from available subscriptions if registration succeeds (bool, default false, requires ansible >= 2.5)
 * `rhsm_environment` - Register with a specific environment in the destination org. (Used with Red Hat Satellite 6 or Katello)
+* `rhsm_pool` - specific pool name to consume (regex can be used as input)
```
Comment (tehsmyers) [inline README.md]: The admonition from the [redhat_subscrition](https://docs.ansible.com/ansible/latest/modules/redhat_subscription_module.html) module docs should probably be included here: "Use pool_ids instead if possible, as it is much faster." Additionally, `rhsm_pools` needs to indicate that it is now mutually exclusive with `rhsm_pool_ids`, and `rhsm_pool_ids` needs to indicate that it is now mutually exclusive with `rhsm_pools`

**Review by tehsmyers — CHANGES_REQUESTED** [review]

**Comment by tehsmyers** [comment]
ok build

**Review by tehsmyers — APPROVED** [review]

**Comment by tehsmyers** [comment]
Passed super-secret downstream CI

**Pull request merged**